operator/src/main/resources/scripts/startServer.sh

```diff
@@ -126,11 +147,95 @@ function copySitCfg() {
   fi
 }
 
+# prepare mii server
+
+function prepareMIIServer() {
+
+  trace "Model-in-Image: Creating domain home."
+
+  # primordial domain contain the basic structures, security and other fmwconfig templated info
+  # domainzip only contains the domain configuration (config.xml jdbc/ jms/)
+  # Both are needed for the complete domain reconstruction
+
+  if [ ! -f /weblogic-operator/introspector/primordial_domainzip.secure ] ; then
+    trace "Domain Source Type is FromModel, the primordial model archive is missing, cannot start server"
+    exit 1
+  fi
+
+  if [ ! -f /weblogic-operator/introspector/domainzip.secure ] ; then
+    trace "Domain type is FromModel, the domain configuration archive is missing, cannot start server"
+    exit 1
+  fi
+
+  cd / && base64 -d /weblogic-operator/introspector/primordial_domainzip.secure > /tmp/domain.tar.gz && \
+   tar -xzf /tmp/domain.tar.gz
+
+  # decrypt the SerializedSystemIni first
+  if [ -f ${RUNTIME_ENCRYPTION_SECRET_PASSWORD} ] ; then
+    MII_PASSPHRASE=$(cat ${RUNTIME_ENCRYPTION_SECRET_PASSWORD})
+  else
+    trace SEVERE "Domain Source Type is 'FromModel' which requires specifying a runtimeEncryptionSecret " \
+    "in your domain resource and deploying this secret with a 'password' key, but the secret does not have this key."
+    exit 1
+  fi
+  encrypt_decrypt_domain_secret "decrypt" ${DOMAIN_HOME} ${MII_PASSPHRASE}
+
+  # restore the config zip
+  #
+  cd / && base64 -d /weblogic-operator/introspector/domainzip.secure > /tmp/domain.tar.gz && \
+    tar -xzf /tmp/domain.tar.gz
+  chmod +x ${DOMAIN_HOME}/bin/*.sh ${DOMAIN_HOME}/*.sh
+
+  # restore the archive apps and libraries
+  #
+  trace "Model-in-Image: Deploying libraries."
+
+  mkdir -p ${DOMAIN_HOME}/lib
+  if [ $? -ne 0 ] ; then
+    trace "Domain type is FromModel, cannot create ${DOMAIN_HOME}/lib "
```

Comment:
Similar... "Domain source type" and "trace SEVERE"

Reply:
done